### PR TITLE
(gh-680)[openstack] enable support for multiple regions

### LINKED ISSUE #680: Openstack support lacks region support
When using beaker with a multi-region Openstack, we need to pass this through to Fog, else it errors with:

    Multiple regions available choose one of these 'emea1,amers1,amers2,amers3'

### PR DESCRIPTION
**Pull request by fiddyspence**
Add optional :openstack_region config key to tell beaker which region to use.

Add support for non-floating IPs, if the machine is provisioned automagically with one.

Fixes #680

### CODE REVIEW

**Comment by anodelman** [comment]
Please associate your commit with a gh issue number like (gh-xxx).

@electrical, how does this look to you?

File: lib/beaker/hypervisor/openstack.rb [hunk 1]
```diff
@@ -115,13 +127,26 @@ def provision
         # Associate a public IP to the server
         # Create if there are no floating ips available
         #
-        ip = @compute_client.addresses.find { |ip| ip.instance_id.nil? }
-        if ip.nil?
-          @logger.debug "Creating IP for #{host.name} (#{host[:vmhostname]})"
-          ip = @compute_client.addresses.create
+        # Do we already have an address?
```
Comment (anodelman) [inline lib/beaker/hypervisor/openstack.rb]: Was this code here before?  This feels like a ride along that should be split out into a separate commit or PR.

File: lib/beaker/hypervisor/openstack.rb [hunk 2]
```diff
@@ -27,25 +28,36 @@ def initialize(openstack_hosts, options)
...
-        :openstack_auth_url => @options[:openstack_auth_url],
-        :openstack_tenant => @options[:openstack_tenant])
+
+      networkoptionhash = {}
+      networkoptionhash[:provider]           = :openstack
+      networkoptionhash[:openstack_api_key]  = @options[:openstack_api_key]
+      networkoptionhash[:openstack_username] = @options[:openstack_username]
+      networkoptionhash[:openstack_auth_url] = @options[:openstack_auth_url]
+      networkoptionhash[:openstack_tenant]   = @options[:openstack_tenant])
+      networkoptionhash[:openstack_region]   = @options[:openstack_region] if @options[:openstack_region],
```
Comment (kevpl) [inline lib/beaker/hypervisor/openstack.rb]: unnecessary line-ending comma here.  copy-paste error?  Otherwise, everything looks clean.  Once this is fixed, I'll be happy to push this through last round testing and getting it merged.

**Comment by anodelman** [comment]
Review comment still current.

**Comment by anodelman** [comment]
Review comments still current, will close out next week due to inactivity.

**Comment by anodelman** [comment]
Closing due to inactivity.

**Comment by anodelman** [comment]
Here's your associated beaker jira bug:

https://tickets.puppetlabs.com/browse/BKR-27
